### PR TITLE
feat: implement shadcn dark mode

### LINKED ISSUE #63: Enhance Dark Mode Implementation
## 📘 Description

- The current dark mode implementation works well but can be improved to simplify maintenance and provide a more customizable experience. By adopting the **shadcn/ui** approach, its components will automatically adjust to dark mode and allow users to choose between their personal preference or the system/browser setting.

## 🔢 Steps

1. Replace the current dark mode logic with the approach outlined in the official shadcn/ui documentation.
2. Ensure that the change is applied correctly in both light and dark modes, according to the user's selected preference.

## ✅ Acceptance Criteria

- The dark mode logic has been successfully replaced with the **shadcn/ui** approach.
- All components and pages work correctly in both light and dark modes according to the user's preference.

## 🌎 References

- [Official shadcn/ui dark mode documentation](https://ui.shadcn.com/docs/dark-mode/next)

## 📜 Additional Notes

- If the filters don’t display correctly in dark mode, don’t worry. These issues will be addressed in future improvements.
- Blocked by: #54 

⚠ **Do not apply until the ODHack begins.**
⚠ **Read our guidelines before applying.**

### PR DESCRIPTION
**Pull request by AnoukRImola**
# 📝 Pull Request Title

## 🛠️ Issue
- Closes #63 

## 📖 Description
- Modification of the current dark mode

## ✅ Changes made
- Removed previous dark mode implementation that used localStorage
- Implemented shadcn/ui theme system with next-themes
- Created new theme provider component
- Updated layout.tsx to use ThemeProvider
- Created new theme toggle component with system preference support

## 🖼️ Media (screenshots/videos)

https://github.com/user-attachments/assets/d9b8be09-8218-4c80-8fdd-f2efa48fcc1a

### Key Files Modified
- `app/components/providers/theme-provider.tsx` (new)
- `app/components/ui/theme-toggle.tsx` (new)
- `app/layout.tsx`
- `app/components/header/Header.tsx`
- `app/components/products/ProductDetailModal.tsx`
- `app/pages/products/[productId]/page.tsx`
- `app/pages/page.tsx`

### Components Updated for Dark Mode
- Replaced static color classes with theme variables:
  - `text-black/white` → `text-foreground`
  - `text-gray-{n}` → `text-muted-foreground`
  - `bg-white/black` → `bg-background`

### Technical Implementation
- Integrated next-themes package for theme management
- Added support for three theme modes:
  - Light
  - Dark
  - System (default)
- Theme persists across page reloads
- Automatic system preference detection

### CODE REVIEW

File: frontend/app/products/[productId]/page.tsx [hunk 1]
```diff
@@ -1,110 +1,115 @@
 import Bounded from "@/app/components/Bounded";
 import SubHeader from "@/app/components/header/subheader/SubHeader";
 import Images from "@/app/components/products/Images";
+import { Button } from "@/app/components/ui/button"; // Cambiado a shadcn/ui
```
Comment (derianrddev) [inline frontend/app/products/[productId]/page.tsx]: remove this comment

**Review by derianrddev — CHANGES_REQUESTED** [review]
@AnoukRImola Please resolve the conflicts

**Review by derianrddev — APPROVED** [review]
@AnoukRImola Nice job!

**Comment by danielcdz** [comment]
@AnoukRImola linda cachorra

**Review by danielcdz — APPROVED** [review]
LGTM

**Pull request merged**